.kokoro/bom-upper-bounds-check.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Presubmit to ensure the dependencies of the Google Libraries BOM, with the modification of change
+# in the PR, pick up the highest versions among transitive dependencies.
+# https://maven.apache.org/enforcer/enforcer-rules/requireUpperBoundDeps.html
+
+set -eo pipefail
+# Display commands being run.
+set -x
+
+cd github/google-cloud-java/
+
+# Print out tool version
+echo $JOB_TYPE
+java -version
+mvn -version
+git version
+
+# Make artifacts available for 'mvn validate' at the bottom
+mvn install -DskipTests=true -Dmaven.javadoc.skip=true -Dgcloud.download.skip=true -B -V
+
+# The property key for this repository in Google Libraries BOM
+VERSION_KEY=google.cloud.java.version
+# Example version: '0.116.1-alpha-SNAPSHOT'
+VERSION_POM=google-cloud-bom/pom.xml
+VERSION=`perl -nle 'print $1 if m|<version>(.+)</version>|' $VERSION_POM |head -1`
```

Comment:
@chingor13 Do we guarantee perl to be installed in the images we use for builds? This is the first I've seen perl used in builds scripts.

Reply:
Good question. We don't explicitly guarantee perl is on our images. It happens to be installed as our base images extended from openjdk which uses a debian image (buildpack-deps:stretch-scm) and has perl installed by default.

Reply:
I agree. We shouldn't be using perl here, and this may be getting complex enough that it jutifies being written with unit tsts. .

Reply:
Let me think a better way than Perl then.

Reply:
Now it uses `xmllint`.